doc/sphinx/grammar.rst

```diff
@@ -0,0 +1,2247 @@
+                    : `fixannot`
+                    : `binder` `binders_fixannot`
+                    : `empty`
+   open_binders : `name` `name_list_opt2` : `lconstr`
+                : `name` `name_list_opt2` `binders`
+                : `name` .. `name`
+                : `closed_binder` `binders`
+   name_list_opt2 : `name_list_opt2` `name`
+                  : `empty`
+   binders : `binder_list_opt`
+   binder_list_opt : `binder_list_opt` `binder`
+                   : `empty`
+   binder : `name`
+          : `closed_binder`
+   typeclass_constraint : ! `operconstr200`
+                        : { `name` } : `exclam_opt` `operconstr200`
+                        : `name_colon` `exclam_opt` `operconstr200`
+                        : `operconstr200`
+   type_cstr : `lconstr_opt2`
+             : : `lconstr`
+             : `empty`
+   preident : `IDENT`
+   pattern_identref : ? `ident`
+   var : `ident`
+   field : `FIELD`
+   fields : `field` `fields`
+          : `field`
+   fullyqualid : `ident` `fields`
+               : `ident`
+   basequalid : `ident` `fields`
+              : `ident`
+   name : _
+        : `ident`
+   reference : `ident` `fields`
+             : `ident`
+   by_notation : `ne_string` `IDENT_opt`
+   IDENT_opt : % `IDENT`
+             : `empty`
+   smart_global : `reference`
+                : `by_notation`
+   qualid : `basequalid`
+   ne_string : `STRING`
+   ne_lstring : `ne_string`
+   dirpath : `ident` `field_list_opt`
+   field_list_opt : `field_list_opt` `field`
+                  : `empty`
+   string : `STRING`
+   lstring : `string`
+   integer : `INT`
+           : - `INT`
+   natural : `INT`
+   bigint : `INT`
+   vernac_control : Time `located_vernac`
+                  : Redirect `ne_string` `located_vernac`
+                  : Timeout `natural` `vernac_control`
+                  : Fail `vernac_control`
+                  : `decorated_vernac`
+   decorated_vernac : `quoted_attributes_list_opt` `vernac`
+   quoted_attributes_list_opt : `quoted_attributes_list_opt` `quoted_attributes`
+                              : `empty`
+   quoted_attributes : #[ `attribute_list` ]
+   attribute_list : `attribute_list_opt`
+   attribute_list_opt : `attribute_list_opt` , `attribute`
+                      : `empty`
+   attribute : `ident` `attribute_value`
+   attribute_value : = `string`
+                   : ( `attribute_list` )
+                   : `empty`
+   vernac : Local `vernac_poly`
+          : Global `vernac_poly`
+          : `vernac_poly`
+   vernac_poly : Polymorphic `vernac_aux`
+               : Monomorphic `vernac_aux`
+               : `vernac_aux`
+   vernac_aux : Program `gallina` .
+              : Program `gallina_ext` .
+              : `gallina` .
+              : `gallina_ext` .
+              : `command` .
+              : `syntax` .
+              : `subprf`
+              : `command_entry`
+   noedit_mode : `query_command`
+   subprf : `BULLET`
```

Comment:
Is this `BULLET` something primitive? Currently, it is defined in the manual at https://coq.inria.fr/refman/proof-engine/proof-handling.html#bullets.

Reply:
`BULLET` is a token.  BTW, it seemed to me that `LEFTQMARK` may not be needed (though could be wrong).